### PR TITLE
GL-264 calculate contamination in gdc somatic wgs pipeline

### PR DESCRIPTION
**Pull request by jessicaway**
Added a calculate contamination task created by the TAG team to calculate the contamination of somatic samples in the GDC whole genome somatic single sample pipeline

### CODE REVIEW

**Comment by jessicaway** [comment]
Final output of an example run can be seen [here](https://cromwell.gotc-dev.broadinstitute.org/api/workflows/1/2f4ec724-67f4-429e-a963-b137d7ccdbcb/outputs
)
Intermediate outputs of the new task are at :
gs://broad-gotc-dev-cromwell-execution/GDCWholeGenomeSomaticSingleSample/2f4ec724-67f4-429e-a963-b137d7ccdbcb/call-check_contamination/contam.txt
gs://broad-gotc-dev-cromwell-execution/GDCWholeGenomeSomaticSingleSample/2f4ec724-67f4-429e-a963-b137d7ccdbcb/call-check_contamination/contamination.table
gs://broad-gotc-dev-cromwell-execution/GDCWholeGenomeSomaticSingleSample/2f4ec724-67f4-429e-a963-b137d7ccdbcb/call-check_contamination/normal_pileups.table
gs://broad-gotc-dev-cromwell-execution/GDCWholeGenomeSomaticSingleSample/2f4ec724-67f4-429e-a963-b137d7ccdbcb/call-check_contamination/pileups.table
gs://broad-gotc-dev-cromwell-execution/GDCWholeGenomeSomaticSingleSample/2f4ec724-67f4-429e-a963-b137d7ccdbcb/call-check_contamination/segments.table

File: beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/GDCWholeGenomeSomaticSingleSample.wdl
Comment (gbggrant) [inline beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/GDCWholeGenomeSomaticSingleSample.wdl:733]: Is this related to the contamination change?  Or just an improvement?

**Review by gbggrant — APPROVED** [review]

File: beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/GDCWholeGenomeSomaticSingleSample.wdl
Comment (jessicaway) [reply beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/GDCWholeGenomeSomaticSingleSample.wdl:733]: It is not related. These are required inputs to the task and were being passed from the inputs file directly to the task. I preferred to pass them to the pipeline inputs and then to the task, since I don't like the concept of having required inputs with no defaults left out of the top level pipeline inputs

**Review by khajoue2 — APPROVED** [review]

File: tasks/broad/CheckContaminationSomatic.wdl
Comment (rexwangcc) [inline tasks/broad/CheckContaminationSomatic.wdl:86]:
```suggestion
}

```

**Review by rexwangcc — COMMENTED** [review]
Hornet has a ticket that's blocked by having:
1. fragment statistics 
2. contamination metrics. 
as the pipeline outputs, I'm not sure if this is the PR adds both of them?

**Comment by jessicaway** [comment]
@rexwangcc The insert size metrics were added by a previous ticket, so this PR should unblock you. We were waiting for a scientific review of the PR, which I will follow up on

**Pull request merged**
